src/Pim/Component/Catalog/Updater/Remover/PriceCollectionAttributeRemover.php

```diff
@@ -85,15 +85,20 @@ protected function removePrices(ProductInterface $product, AttributeInterface $a
     {
         $productValue = $product->getValue($attribute->getCode(), $locale, $scope);
 
+        $currencyToRemove = [];
+        foreach ($data as $priceToRemove) {
+            $currencyToRemove[] = $priceToRemove['currency'];
+        }
+
         if (null !== $productValue) {
             $prices = [];
             foreach ($productValue->getData() as $price) {
-                if (!in_array(['amount' => $price->getData(), 'currency' => $price->getCurrency()], $data)) {
```

Comment:
That is a PO question. If the collection contains 4 € and you ask to remove 5 €, what should happen?